feathr_project/feathr/_feature_registry.py

```diff
@@ -838,3 +839,142 @@ def _list_registered_entities_with_details(self, project_name: str = None, entit
+        feature_list=[]
+        key_list = []
+        for feature_entity in feature_entities:
+            for key in feature_entity["attributes"]["key"]:
+                key_list.append(TypedKey(key_column=key["key_column"], key_column_type=key["key_column_type"], full_name=key["full_name"], description=key["description"], key_column_alias=key["key_column_alias"]))
+
+            # after get keys, put them in features
+            feature_list.append(Feature(name=feature_entity["attributes"]["name"],
+                    feature_type=None, # stored as a hocon string, can be parsed using pyhocon
+                    transform=None, #transform attributes are stored in a dict fashion , can be put in a WindowAggTransformation
+                    key=key_list,
+                    registry_tags=feature_entity["attributes"]["tags"],
+
+            ))
+        return feature_list 
+    def get_feature_by_fqdn_type(self, qualifiedName, typeName):
+        """
+        Get a single feature by it's QualifiedName and Type
+        Returns the feature else throws an AtlasException with 400 error code
+        """
+        response = self.purview_client.get_entity(qualifiedName=qualifiedName, typeName=typeName)
+        entities = response.get('entities')
+        for entity in entities:
+            if entity.get('typeName') == typeName and entity.get('attributes').get('qualifiedName') == qualifiedName: 
+                return entity
+       
+    def get_feature_by_fqdn(self, qualifiedName):
+        """
+        Get feature by qualifiedName
+        Returns the feature else throws an AtlasException with 400 error code
+        """        
+        guid = self.get_feature_guid(qualifiedName)
+        return self.get_feature_by_guid(guid)
+    
+    def get_feature_by_guid(self, guid):
+        """
+        Get a single feature by it's GUID
+        Returns the feature else throws an AtlasException with 400 error code
+        """ 
+        response = self.purview_client.get_single_entity(guid=guid)
+        return response
+    
+    def get_feature_lineage(self, guid):
+        """
+        Get feature's lineage by it's GUID
+        Returns the feature else throws an AtlasException with 400 error code
+        """
+        return self.purview_client.get_entity_lineage(guid=guid)
+
+    def get_feature_guid(self, qualifiedName):
+        """
+        Get guid of a feature given its qualifiedName
+        """        
+        search_term = "qualifiedName:{0}".format(qualifiedName)
+        print(search_term)
+        entities = self.purview_client.discovery.search_entities(search_term)
+        for entity in entities:
+            if entity.get('qualifiedName') == qualifiedName:
+                return entity.get('id')
+
+    def search_features(self, searchTerm):
+        """
+        Search the registry for the given query term
+        For a ride hailing company few examples could be - "taxi", "passenger", "fare" etc.
```

Comment:
example should start with "example" or "for example" so the users know it's example

Reply:
The example is not complete. Please give the return results for these key words and criteria why it's returned.

Reply:
Search functions will be done later, resolve for now